### DISCUSSION TITLE
[FR] Update repository minor/patch version on font updates

### DISCUSSION THREAD

**Discussion by quinnjr** [ryanoasis/nerd-fonts#771] [Ideas]
Hello,

I maintain a few of the `nerd-font` packages in the Arch Linux AUR and would like to see this repo release new versions whenever the fonts in the collection are updated.

In particular, Fira Code has been updated since the last release (v2.1.0), but no minor bumps or patch versions have been released for the entirety of the repo to reflect this.

My packages stick to the latest releases of the `nerd-fonts` package as a whole. It would be beneficial for my ability to keep the packages updated if I could receive Github notifications on updates to the fonts instead of having to keep tabs on each font's updates as changes are brought in from the non-patched versions.

Thank you @ryanoasis

**Comment by LlsaintA1** [comment]
> Hello,
> 
> I maintain a few of the `nerd-font` packages in the Arch Linux AUR and would like to see this repo release new versions whenever the fonts in the collection are updated.
> 
> In particular, Fira Code has been updated since the last release (v2.1.0), but no minor bumps or patch versions have been released for the entirety of the repo to reflect this.
> 
> My packages stick to the latest releases of the `nerd-fonts` package as a whole. It would be beneficial for my ability to keep the packages updated if I could receive Github notifications on updates to the fonts instead of having to keep tabs on each font's updates as changes are brought in from the non-patched versions.
> 
> Thank you @ryanoasis

Kyle D Puskas

**Reply by Finii** [reply]
What is the meaning of this comment? :thinking: Sorry I do not understand.

**Comment by Finii** [comment]
Dear @quinnjr

it would indeed be great if we wound automatically create a new patched version every time one sourcefont is updated.
But I guess that is hard to implement with the current Nerd Fonts release model, which is also related to how this whole repository is set up and is used. It just does not fit.

The only thing I (who is just a maintainer with limited rights) can do is to release more often. This is not as good as automatic release on source update (which we have done at Delugia for Cascadia Code only), but at least it will/can help to minimize the gap between current source font and latest patched version.

Indeed with `v2.2.0` the aim was to enable more frequent and automatic releases. But there were so many bugs still in the patcher that I thought we first need to get the patching into shape and work on updates/releases afterwards. With the almost ready `v2.3.0` I believe the biggest kinks have been smoothed out in the code, and we can at least manually update all source fonts. In fact I prepared a list https://github.com/ryanoasis/nerd-fonts/discussions/962#discussioncomment-3894678 of pending source updates.

If we really want to go into automatic updates we need to split this repo in 'the patcher script' and 'the patched fonts', or even better one repo (with individual releases and version numbers) for each source/patched font. But while there is already the patcher repo (albeit empty) at https://github.com/nerdfonts I can not create new repos under that organization - which would be needed. Unfortunately Ryan is here only very intermittently anymore.

You mention Fira specifically. So... If you look at https://github.com/ryanoasis/nerd-fonts/discussions/962 you see `v2.3.0` will update (only) 3270, Inconsolata, and Iosevka. If you tell me Fira is especially important we can of course add that update also to `v2.3.0`, which will delay that release a bit. The other possibility is to wait for `2.3.1` or `3.0.0` which then will update all the remaining source fonts. As there are no code changes planned for that release it could come out rather quickly.

So, in summary you are right; but I can not solve this and Ryan is unlikely to hear that. We can only do what we can do, sorry.

Looking forward to your reply :-)

----

But as you are an AUR maintainer, I have a question. I just noticed that

https://aur.archlinux.org/packages/nerd-fonts-fira-code/

seems to have vanished, and I can only find

https://aur.archlinux.org/packages/otf-nerd-fonts-fira-code

The package names at AUR seem to not be the most stable? If the AUR maintainers would inform us here we could update the links here.
AUR is just one of the distribution channels (see https://github.com/ryanoasis/nerd-fonts/discussions/881#discussioncomment-3488054) and the mere number of ways people can get the Nerd Fonts makes any change very very hard.

**Reply by quinnjr** [reply]
This ticket is a bit old. The `nerd-fonts-fira-code` package has since been moved from the AUR to the Community repository.

**Reply by Finii** [reply]
Thanks. Still would be good to get some info on that.

I also noticed that we already updated to FiraCode 6.2 (i.e. latest), and it has been released with `2.2.0` (as you say: old ticket).
But the problem is still there of course. The problem that we do not auto-update.

Just wanted to squeeze the update in, and noticed that the files do not change :-D
`v2.3.0` is likely to come out in the next days.

**Reply by quinnjr** [reply]
@Finii I just actually had a number of the fonts I was maintaining get promoted to [Community]. I'll need to get a list together of them and update installation procedures.

**Reply by Finii** [reply]
Great!
But then, did you see #1057 that suggests not listing the archives individually anymore?
What is your opinion?

I actually thought about putting a complete list of Casks somewhere for people that use Homebrew somewhere.
From a user perspective it might be nicer to have a real concrete list then a link to a search engine.

Homebrew, because for these I wrote a workflow here that updates them all (not that I myself ever used Homebrew).

The Cask names are also available already (but not displayed) on the web-pages (www.nerdfonts.com) and one could in principle add them to the font list there. Otoh I'd rather not drive all the packaging systems from here, that would be a fulltime job on its own.

**Reply by quinnjr** [reply]
@Finii I'd probably drop them as well and just add a search for the community repository for the fonts. I'm not sure if all of them have been migrated, though. I know mine were because they were `ttf`
